src/freenas/etc/ix.rc.d/ix-crontab

```diff
@@ -75,6 +75,22 @@ generate_crontab()
 		echo "${scrub_minute}	${scrub_hour}	${scrub_daymonth}	${scrub_month}	${scrub_dayweek}	root	PATH=\"$cron_PATH\" /usr/local/libexec/nas/scrub -t ${scrub_threshold} ${vol_name}" >> /etc/crontab
 	done
 
+	local f="begin end"
+	eval local $f
+	local sf=$(var_to_sf $f)
+	local begin_hour begin_minute end_hour end_minute
+	# Generate resilver priority
+	${FREENAS_SQLITE_CMD} -separator "" ${RO_FREENAS_CONFIG} \
+	"SELECT $sf FROM storage_resilver WHERE enabled = 1 ORDER BY -id" | \
+	while eval read -r $f; do
+		begin_hour=$(echo ${begin}|cut -d: -f 1)
+		begin_minute=$(echo ${begin}|cut -d: -f 2)
+		end_hour=$(echo ${end}|cut -d: -f 1)
+		end_minute=$(echo ${end}|cut -d: -f 2)
+		echo "${begin_minute}	${begin_hour}	*	*	*	root	/usr/local/bin/midclt call pool.configure_resilver_priority > /dev/null 2>&1" >> /etc/crontab
+		echo "${end_minute}	${end_hour}	*	*	*	root	/usr/local/bin/midclt call pool.configure_resilver_priority > /dev/null 2>&1" >> /etc/crontab
+	done
+
```

Comment:
Spoke with @william-gr about this and he said he will look into it tomorrow, but basically here is my concern (I may just be wrong but wanted to walk you'll through what I am thinking):

- The user selects weekdays of Monday and Thursday thus, resilver objects `weekday` var will contain `'1, 4'`

- She/He, however, selects a begin of 9 pm in the evening and end time of 6 am

- Now, the cron job looks as follows:
```
0    21    *    *    *        root    /usr/local/bin/midclt call pool.configure_resilver_priority
0     6    *    *    *        root    /usr/local/bin/midclt call pool.configure_resilver_priority
```
- On Monday, at 9pm i.e 2100 hours, `pool.configure_resilver_priority` is called and it sets the higher priority resilvering sysctls
- On Tuesday, at 12.30 am i.e. 0030 hours, the system (due to any unforeseen or planned reasons) restarts and at this point, middlewared is started, which loads all its plugins and when the pool plugin is loaded its `setup` routine will reset the sysctl's to lower priority ones as the following condition is not met:
```
now = datetime.now()
if now.isoweekday() in weekdays:
```
Now, since resilvering will still continue to post the restart (since it is a state of the pool and zfs ensures that they resume), we will now have a 5 and a half hour period (from 12.30 am to 6 am) where the resilvering should have happened at higher priority but did not.

My conclusion may be flawed, in which case please elaborate on how it is so, so that I can understand where I went wrong.

In the event that my conclusion is correct, there are 2 outcomes as I see it:
1. This is an edge case and we should not convolute the code that runs 95% of the time for the other 5% scenario.
2. We handle this exception via some `if` and `else`'s or some such in `pool.configure_resilver_priority` call

Thanks.